### PR TITLE
Imag view/init

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Viewing utility.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
This is basically ready, though features are missing. I'd merge it and implement features later on.

File: imag-view/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                  rt: &'a Runtime)
+    -> StoreResult<FileLockEntry<'a>>
+{
+    debug!("Checking path element for version");
+
+    let version = {
+        version.unwrap_or_else(|| {
+            id.split("~").last().unwrap_or_else(|| {
+                warn!("No version");
+                exit(1);
```
Comment (TheNeikos) [inline imag-view/src/main.rs]: I am not sure about `exit` it basically does not allow destructors to run, is that something we can afford?

**Comment by TheNeikos** [comment]
Except what I annotated :+1:

File: imag-view/src/main.rs (hunk 1 above)
Comment (matthiasbeyer) [inline imag-view/src/main.rs]: I will fix this. This will require me to add a `error` module, so we can wrap the things and trace appropriately.

**Comment by matthiasbeyer** [comment]
Will rebase onto lastest master, which will shorten this PR a bit.

**Comment by matthiasbeyer** [comment]
@TheNeikos please reiterate.

File: imag-view/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    let about = "View entries (readonly)";
+    let ui = build_ui(Runtime::get_default_cli_builder(name, version, about));
+    let rt = {
+        let rt = Runtime::new(ui);
+        if rt.is_ok() {
+            rt.unwrap()
+        } else {
+            println!("Could not set up Runtime");
+            println!("{:?}", rt.err().unwrap());
+            exit(1); // we can affort not-executing destructors here
```
Comment (TheNeikos) [inline imag-view/src/main.rs]: with a d

File: imag-view/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    debug!("Lets get rollin' ...");
+
+    info!("No implementation yet");
+
+    let entry_id = rt.cli().value_of("id").unwrap(); // enforced by clap
+
+    if rt.cli().is_present("versions") {
+        if let Err(e) = view_versions_of(entry_id, &rt) {
+            trace_error(&e);
+            exit(1);
```
Comment (TheNeikos) [inline imag-view/src/main.rs]: Same reasoning as above I presume?

File: imag-view/src/main.rs (hunk 2 above)
Comment (matthiasbeyer) [inline imag-view/src/main.rs]: I guess I wanted to write "effort" ...

File: imag-view/src/main.rs (hunk 3 above)
Comment (matthiasbeyer) [inline imag-view/src/main.rs]: Yes, though here it is a clean exit with status 1. How would you do that? The program would exit after `trace_error()` anyways, but I want to exit with `EXIT_FAILURE` here...

Comment (TheNeikos) [inline imag-view/src/main.rs]: I ask because you did not add a comment :)

**Pull request merged**